17-lifecycle-lab/src/02-cps-with-indicator.jsx

```diff
@@ -1,26 +1,60 @@
-/**
- * Lab 02: CPS monitor with indicator
- *
- * Add an indicator panel to the CPS monitor
- * that shows a green rectangle if user clicks fast enough
- * or a red rectangle if user clicks too slow
- *
- * Decide how to share information between the components
- * and what values to pass when creating the Indicator
- */
-
-import React from 'react';
-
-const Indicator = React.createClass({
-});
-
-export default createClass({
-  render() {
-    return (<div>
-      <button>Click Fast</button>
-      <p>CPS rate: {0}</p>
-      <Indicator />
-    </div>);
-  }
-});
-
+import React from 'react';
+
+export default class ClicksCounter extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            counter: 0,
+            fastPlayer: false,
+        };
+
+        this.incClickCounter = this.incClickCounter.bind(this);
+    }
+
+    componentWillMount() {
+        this.timer = setInterval(() => {
+            this.setState({counter: 0});
+        }, 1000);
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timer);
+    }
+
+    incClickCounter() {
+        let counter = this.state.counter + 1;
+        let fastPlayer = false;
+        if (counter > 4) {
+            fastPlayer = true;
+        }
+
+        this.setState({
+            counter,
+            fastPlayer,
+        });
+    }
+
+    render() {
+        return (<div>
+            <button onClick={this.incClickCounter}>Click Fast</button>
+            <div>yo clicked: {this.state.counter}</div>
+            <ColorIndicator fastPlayer={this.state.fastPlayer}/>

```

Comment:
I like the fact that you pass predicate down to the indicator and indicator is the only one that determines the color of the div.